source/configuration_manual/authentication/oauth2.rst

```diff
@@ -253,60 +267,91 @@ Oauth2 overrides some of the default HTTP client and SSL settings:
 You can override these and any other HTTP client or SSL settings by placing
 them inside :dovecot_core:ref:`oauth2` named filter.
 
-.. code-block:: none
 
-  ### OAuth2 password database configuration
+.. dovecot_core:setting:: oauth2_tokeninfo_url
+   :values: @string
+
+   URL for verifying token validity. Token is appended to the URL. Example:
+   ``oauth2_tokeninfo_url = http://endpoint/oauth/tokeninfo?access_token=``
+
+.. dovecot_core:setting:: oauth2_introspection_url
+   :values: auth, get, post, local
+
+   How introspection is made, valid values are
+
+   auth
+     GET request with Bearer authentication
+   get
+     GET request with token appended to URL
+   post
+     POST request with token=bearer_token as content
+   local
+     Attempt to locally validate and decode JWT token
+
+   There is no default value, you must explicitly choose one.
```

Comment:
There is always some default visible in doveconf output - it's not going to give error if it's not explicitly said.. Is it "none" or something? Add it to :default: and clarify here that it must be chosen when oauth2 is used.

Reply:
Ah, it's empty. I hadn't even thought that enum value could be empty. I guess it's clear enough this way.

Reply:
Or .. is this the setting the actually enables oauth2, and the rest of the settings are ignored if this is left empty? In that case, that should be clarified here.

Reply:
yeah, the point is that if it's set to non-empty value then settings checks fail.

Reply:
actually the code treats empty as auth.